### ISSUE TITLE
separation of concerns for top level classes and parsers

### ISSUE THREAD

**Issue by Jaymon** [Jaymon/plain#2]
it would probably make more sense for the url portions to be in the plain level classes, not the plain.parsers level classes, that way the parsers can just focus on the actual parsing and the top level classes handle all the downloading and stuff, the root level classes can better determine the interface that people will use (is a url even needed?) and will make it so the top level classes aren't just thin wrappers around the parsers which basically handle everything

Basically, it would make sense for the functionality to be split between the top level classes (which is the public api external codebases will use) and the parsers (which are more internal and should be focused on, you know, parsing).

**Comment by Jaymon** [comment]
This is what #6 is talking about also, these can probably be closed at the same time
